### PR TITLE
Add Scott's GPU-based Coral Image Analysis code as a correctness and performance test

### PR DESCRIPTION
**Pull request by e-kayrakli**
This adds the un-optimized code as a correctness and performance test. I am planning to add the optimizations later on to record the impact.

### CODE REVIEW

**Comment by e-kayrakli** [comment]
@sdbachman -- Could you take a look when you get the chance? I didn't change much, but added a simple checksum for a very basic correctness check. If you have a better way for testing validity of the output that'd be great.

**Comment by sdbachman** [comment]
I don't have a better way of doing the correctness check, but the code all looks good to me.

**Comment by e-kayrakli** [comment]
@DanilaFe -- I asked for a review, but the actual application code is something Scott and I have been working on. For the time being, it'd be nice to stick to its original form as much as possible. If you see any glaring mistakes, definitely let me know. But I'd be a bit reluctant to change the way it is implemented from the initial PR.

File: test/gpu/native/studies/coral/coral.chpl [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        use Random;
+        fillRandom(Array, seed=13);
+        if write_data then writeln(Array);
+      }
+
+      // Create Block distribution of interior of PNG
+      const offset = nx+1; // maybe needs to be +1 to account for truncation?
+      const Inner = ImageSpace.expand(-offset);
+      const myTargetLocales = reshape(Locales, {1..Locales.size, 1..1});
+      const D = Inner dmapped Block(Inner, targetLocales=myTargetLocales);
```
Comment (DanilaFe) [inline test/gpu/native/studies/coral/coral.chpl:161]: I don't see this used anywhere -- is this a carry-over from the non-GPU implementation? Should we delete this code, along with the `reshape` operation above?

**Review by DanilaFe — APPROVED** [review]

File: test/gpu/native/studies/coral/coral.chpl (hunk 1 above)
Comment (e-kayrakli) [reply test/gpu/native/studies/coral/coral.chpl:161]: Good point. How about I keep it as commented with some description?

Your assessment is probably correct, but we also want to have a similar thing going forward where we modify this to be multilocale and multi GPU.

**Pull request merged**